app/src/main/res/layout/fragment_credit_card_editor.xml

```diff
@@ -0,0 +1,237 @@
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:layout_marginEnd="24dp"
+            android:orientation="vertical">
+
+            <androidx.appcompat.widget.AppCompatSpinner
+                android:id="@+id/expiry_month_drop_down"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                tools:listitem="@android:layout/simple_list_item_1" />
+
+            <View
+                android:layout_width="match_parent"
+                android:layout_height="1dp"
+                android:background="?primaryText" />
+
+        </LinearLayout>
+
+        <LinearLayout
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:orientation="vertical">
+
+            <androidx.appcompat.widget.AppCompatSpinner
+                android:id="@+id/expiry_year_drop_down"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                tools:listitem="@android:layout/simple_list_item_1" />
+
+            <View
+                android:layout_width="match_parent"
+                android:layout_height="1dp"
+                android:background="?primaryText" />
+
+        </LinearLayout>
+    </LinearLayout>
+
+    <!-- Card Nickname -->
+    <TextView
+        android:id="@+id/card_nickname_title"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:gravity="center_vertical"
+        android:paddingStart="3dp"
+        android:paddingEnd="0dp"
+        android:text="@string/credit_cards_card_nickname"
+        android:textColor="?primaryText"
+        android:textSize="12sp"
+        android:layout_marginTop="10dp"
+        android:letterSpacing="0.05"
+        app:fontFamily="@font/metropolis_semibold" />
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/card_nickname_layout"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:paddingBottom="11dp"
+        android:textColor="?primaryText"
+        app:errorEnabled="true"
+        app:hintEnabled="false">
+
+        <com.google.android.material.textfield.TextInputEditText
+            android:id="@+id/card_nickname_input"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:ellipsize="end"
+            android:fontFamily="sans-serif"
+            android:letterSpacing="0.01"
+            android:lineSpacingExtra="8sp"
+            android:maxLines="1"
+            android:singleLine="true"
+            android:textColor="?primaryText"
+            android:textSize="16sp" />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
Seeing a `ConstraintLayout` in the middle of a layout makes me wonder why it isn't the parent.
But if using just one more layer helps the layout to work and reason with, all good.

Reply:
Ya, I was expecting a comment on this. I considered making everything use ConstraintLayout after finding the need to use it for the buttons. I think we can still make this change, but perhaps as a follow up.